timemachine/fe/rbfe.py

```diff
@@ -408,21 +431,25 @@ def estimate_relative_free_energy(
         warnings.warn("Warning: setting lambda_schedule manually, this argument may be removed in a future release.")
 
     temperature = DEFAULT_TEMP
-    all_frames, all_boxes, all_U_fns = generate_samples(
-        single_topology, host_config, temperature, lambda_schedule, n_frames, seed
-    )
-    all_dGs, all_errs, plot_buffer = estimate_free_energy_given_samples(
-        all_frames, all_boxes, all_U_fns, temperature, lambda_schedule, prefix
-    )
+    initial_states = setup_initial_states(single_topology, host_config, temperature, lambda_schedule, seed)
+    protocol = SimulationProtocol(n_frames=n_frames, burn_in=10000, steps_per_frame=1000)
+
+    try:
+
+        if keep_idxs is None:
+            keep_idxs = [0, -1]  # keep first and last frames
+
+        assert len(keep_idxs) <= len(lambda_schedule)
+
+        prefix = get_mol_name(mol_a) + "_" + get_mol_name(mol_b) + "_" + prefix
+
+        return estimate_free_energy_given_initial_states(initial_states, protocol, temperature, prefix, keep_idxs)
+
+    except Exception as e:
```

Comment:
Seems cleaner to push the exception handling into the `estimate_free_energy_given_initial_states` as it returns a SimulationResult, so it would seem like it could set the exception field.

Reply:
this was de-boggled! all of the exception handling is done on the client side